### ISSUE TITLE
Unifi Protect App: Live view works on WAN but not LAN

### ISSUE THREAD

**Issue by acortelyou** [p10tyr/rtsp-to-onvif#11]
I am currently investigating an issue where pulling up the live view on the phone app works while I'm on WAN (so via unifi cloud presumably) but does not load when on LAN.

This only seems to happen on the phone app in live view. Recordings/timeline are unaffected and load correctly.

Browser on the phone and laptop both work when connected via unifi cloud or direct IP.

Not quite sure why this would be happening unless the phone app is being too clever for its own good and attempting a direct connection when on LAN.

Maybe a problem with multiple concurrent streams?

Will let you know if I sort it out and please let me know if you have seen similar or have ideas.

Thanks.

**Comment by p10tyr** [comment]
It is quite possible that when you are on LAN the Protect app is telling the phone to pull the stream direct to your device over RTSP. If it works over LAN it means protect is proxying it for you.

I am not quite sure why the stream is not loading.. it could be something todo with the virtual IP's on the same network that just dont work in your case locally.. or on the device? 

What device is it?

**Comment by eoprede** [comment]
Yes, the phone Protect app does certainly have some smartness to it.
I have a crappy Amcrest camera (ASH43). If I set it to H264 codec, it does not send a good stream - frames a missing, quality is weird, etc. If I leave it at H264, I can occasionally connect it to UNVR and Protect. I can then watch it from my phone either on LAN or WAN, I can view live stream via browser, but the recordings are broken - simply black screen and nothing else. And the stream disconnects sometimes.
I can also set camera to H265, which it streams way better - no missing frames or any other quality issues. However then, the only way I can watch the camera live stream is on my phone from LAN. I also can watch past recordings, but also only on my phone and while on LAN. So I assume cloud forwarding doesn't work for H265, web browser page also doesn't support it, but phone somehow handles it just fine. I can also download the fragment of the video via the web or the phone and it plays just fine.
Oh and another interesting piece of info - if I used protect app outside and then came home, streaming from the problematic camera won't work either. It will start working in a few minutes - so I am guessing phone app tries local connection occasionally and if it fails - it won't try it again for a while.
